### DISCUSSION TITLE
How to use sksurv.compare.compare_survival in practice?

### DISCUSSION THREAD

**Discussion by yanisvdc** [sebp/scikit-survival#452] [Q&A]
Dear all, I have a crucial question for my research project.

I have randomized clinical trial data:
- binary treatment (treatment group vs control group)
- binary outcome (death=1, alive=0) 
- covariates (age, smoker, alcohol use, etc.)

I have 5 years of data with censoring (patients die or stop following up). I want to assess the heterogeneity of treatment effect on the outcome given the covariates. 

How can I do that?

My understanding is that I need to estimate some hazard metric, I would like to do so non-parametrically - my current choice would be using Random Survival Forests or Gradient Boosted Models. I believe that I will get a hazard metric for each patient in my X_test. 

But then, I am not sure how to conclude. Should I use sksurv.compare.compare_survival? I do not understand how to apply this function for my use case: I understand the parameters:
-  y: structured array: (status, time in days)
- group_indicator: in my example, it would be whether each patient is in the treatment or the control group, right?

Question: Where is the hazard metric calculated using Random Survival Forests or Gradient Boosted Models defined in this function? Also, do I get to choose the number of group-specific harzard metric?

Are there any other tests available in the library that I could use?

I would be infinitely grateful for any help,

Thanks a lot

**Comment by sebp** [comment]
The logarithm of the hazard ratio is the `coef_` attribute of the semi-parametric [sksurv.linear_model.CoxPHSurvivalAnalysis](https://scikit-survival.readthedocs.io/en/latest/api/generated/sksurv.linear_model.CoxPHSurvivalAnalysis.html#sksurv.linear_model.CoxPHSurvivalAnalysis). If you also require p-values and confidence intervals, you have to use other libraries such as statsmodels or lifelines.

Gradient Boosted Models are also semi-parametric, but allow to estimate complex non-linear relationships with survival. A Random Survival Forest is truly non-parametric. Because Gradient Boosted Models and Random Survival Forest allow for complex relationships, they do not allow estimating a hazard ratio as a single number. The hazard ratio only has a meaningful interpretation for the Cox model.
